manifests/src/update_etags.sh

```diff
@@ -1,12 +1,26 @@
 #!/usr/bin/env bash
-# Returns the updated ETag for elements in the second column of $1 alongside the first two columns.
-cat $1 |
- tail -n +2 | # Remove headers
- cut -f2 -d',' | # Select url column
- xargs -I {} -- curl -I --silent "{}" | # Fetch remote metadata 
- grep "ETag" |  # Select etag field from resulting html
- awk '{print $2}' | # Remove prefix
- sed 's/\r$//' | # Remove carriage
- sed 1i'"etag"' | # add header
- paste - $1 -d',' | # Merge with original file
- awk -F ',' '{print $2","$3","$1}' # Print in the right order
+# Fetch updated ETag values for URLs in a CSV file.
+
+input_file="$1"
+
+# Check if input file is provided
+if [ -z "$input_file" ]; then
+  echo "Usage: $0 <input_file>"
+  exit 1
+fi
+
+# Extract URLs using csvkit (assuming URL column is named 'url')
+urls=$(csvcut -c url "$input_file" | tail -n +2)
+
+# Fetch ETags for each URL in a loop to avoid xargs command length issues
```

Comment:
Is the xargs command length a potential issue? I assumed it unlikely our s3 urls could grow that long.

Reply:
Indeed, I got `xargs: command line cannot be assembled, too long`